app/migrations/Version20171227163801.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Migrations;
+
+use Doctrine\DBAL\Migrations\AbstractMigration;
+use Doctrine\DBAL\Schema\Schema;
+
+class Version20171227163801 extends AbstractMigration
+{
+    public function up(Schema $schema)
+    {
+        $this->addSql('UPDATE pages SET slug=\'emmanuel-macron\' WHERE slug= \'emmanuel-macron-ce-que-je-suis\'');
```

Comment:
You must also handle 301s. See https://github.com/EnMarche/en-marche.fr/blob/281f460f0f5425c27787642d0bd0426381006120/src/Redirection/LegacyRedirectionsSubscriber.php#L35.

Reply:
I don't need to do redirection, I change the slug for the `findOneBySlug`. Look changes on `PageController.php`

Reply:
Why did you change all the slugs in that case ? I think I am missing something..

Reply:
It is to simplify the logic inside `SitemapFactory.php`, that's it ?

Reply:
I change all slugs because they are all wrong in database. And I need to change slug to generate the right URL in `SitemapFactory.php`.

Reply:
I don't understand your answer, the Uris did change right?

Reply:
I mean a changed uri should now respond a 404?

Reply:
If "yes" is the answer to my questions above, you MUST handle 301.

Reply:
No URLs are still the same ;-)

Reply:
Ok, got it, thx!